### PR TITLE
[Transaction] Fix broker init transaction related topic.

### PR DESCRIPTION
**Pull request by congbobo184**
## Motivation
now, transaction coordinator assign topic and pending ack and transaction log will init persistent topic when namespace unload, so we should load namespace bundle can't init the above topic.
## implement
load namespace bundle filter related managedLedger topic.
### Verifying this change
Add the tests for it

Does this pull request potentially affect one of the following parts:
If yes was chosen, please highlight the changes

Dependencies (does it add or upgrade a dependency): (no)
The public API: (no)
The schema: (no)
The default values of configurations: (no)
The wire protocol: (no)
The rest endpoints: (no)
The admin cli options: (no)
Anything that affects deployment: (no)

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@congbobo184 does this affect docs?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+@Test(groups = "broker")
+public class TransactionTest extends TransactionTestBase {
+
+    private static final String TENANT = "tnx";
+    private static final String NAMESPACE1 = TENANT + "/ns1";
+
+    @BeforeMethod
+    protected void setup() throws Exception {
+        this.setBrokerCount(1);
+        this.
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java:59]: nit: please fix indent

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        admin.tenants().createTenant(TENANT,
+                new TenantInfoImpl(Sets.newHashSet("appid1"), Sets.newHashSet(CLUSTER_NAME)));
+        admin.namespaces().createNamespace(NAMESPACE1);
+
+        admin.tenants().createTenant(NamespaceName.SYSTEM_NAMESPACE.getTenant(),
+                new TenantInfoImpl(Sets.newHashSet("appid1"), Sets.newHashSet(CLUSTER_NAME)));
+        admin.namespaces().createNamespace(NamespaceName.SYSTEM_NAMESPACE.toString());
+        admin.topics().createPartitionedTopic(TopicName.TRANSACTION_COORDINATOR_ASSIGN.toString(), 1);
+
+        pulsarClient = PulsarClient.builder()
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java:74]: we are creating a new PulsarClient here, we should close the existing one

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            } catch (Exception e) {
+                return false;
+            }
+            return true;
+        });
+
+        admin.namespaces().unload(NamespaceName.SYSTEM_NAMESPACE.toString());
+        admin.namespaces().unload(NAMESPACE1);
+
+        consumerBuilder.subscribe();
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java:111]: one question:
is it generally allowed to use the same ConsumerBuilder twice ?
I believe this is a bad practice, but I am not sure we are using this technique in other tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 4]
```diff
@@ -991,7 +994,11 @@ public void loadNamespaceTopics(NamespaceBundle bundle) {
             for (String topic : getNamespaceService().getListOfPersistentTopics(nsName).join()) {
                 try {
                     TopicName topicName = TopicName.get(topic);
-                    if (bundle.includes(topicName)) {
+                    if (bundle.includes(topicName)
+                            && !topic.startsWith(TopicName.TRANSACTION_COORDINATOR_ASSIGN.toString())
+                            && !topic.startsWith(TopicName.get(TopicDomain.persistent.value(),
+                            NamespaceName.SYSTEM_NAMESPACE, TRANSACTION_LOG_PREFIX).toString())
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1000]: nit: indent

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 5]
```diff
@@ -991,7 +994,11 @@ public void loadNamespaceTopics(NamespaceBundle bundle) {
             for (String topic : getNamespaceService().getListOfPersistentTopics(nsName).join()) {
                 try {
                     TopicName topicName = TopicName.get(topic);
-                    if (bundle.includes(topicName)) {
+                    if (bundle.includes(topicName)
+                            && !topic.startsWith(TopicName.TRANSACTION_COORDINATOR_ASSIGN.toString())
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:998]: does it make sense to create a static method with this list of tests ?

`private static boolean isTransactionSystemTopic(TopicName topicName) ...`

**Review by eolivelli — CHANGES_REQUESTED** [review]
interesting case.
I left a few comments.

PTAL

**Comment by congbobo184** [comment]
@eolivelli  fix your comment. please review again. thanks.

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by congbobo184** [comment]
> @congbobo184 does this affect docs?

don't need.

**Pull request merged**